rust/connlib/tunnel/src/peer.rs

```diff
@@ -167,6 +153,106 @@ impl ClientOnGateway {
         [IpAddr::from(self.ipv4), IpAddr::from(self.ipv6)]
     }
 
+    pub(crate) fn refresh_translation(
```

Comment:
If you model this component with `handle_timeout`, you might be able to write a much simpler `proptest-state-machine` test that takes into account time as well and ensures that the NAT table expires correctly.

The difficulty with time-based tests is correctly modelling, what expires when. But in the scope of just the NAT table, this might be easy enough?